File: jobs/2026-04-25__23-53-07/hatchet_workflow_cancellation__3z7CwPD/artifacts/code/worker.py
#!/usr/bin/env python3
import time
from hatchet_sdk import Hatchet, Context

hatchet = Hatchet()

@hatchet.workflow(name="cancellation_workflow", on_events=["cancellation:create"])
class CancellationWorkflow:
    @hatchet.step()
    def check_flag(self, ctx: Context):
        for i in range(10):
            if ctx.exit_flag:
                print("Task has been cancelled")
                raise ValueError("Task has been cancelled")
            
            print(f"Iteration {i}")
            time.sleep(1)
        
        return {"error": "Task should have been cancelled"}

def main():
    worker = hatchet.worker("cancel-worker")
    worker.register_workflow(CancellationWorkflow())
    worker.start()

if __name__ == "__main__":
    main()

File: jobs/2026-05-26__11-38-52/linear_dag_ts__yp5Gj54/artifacts/user/myproject/src/worker.ts
import { hatchet, workflow } from './workflow.js';

const runId = process.env.ZEALT_RUN_ID || 'default';
const workerName = `linear-dag-worker-${runId}`;

async function main() {
  const worker = await hatchet.worker(workerName, {
    workflows: [workflow]
  });
  await worker.start();
}

main().catch(console.error);
